ocaml/xapi/xapi_vm.ml

```diff
@@ -256,14 +256,36 @@ let start ~__context ~vm ~start_paused ~force =
   in
   let vmr =
     if platform <> vmr.API.vM_platform then begin
-      Db.VM.set_platform ~__context ~self:vm ~value:platform;
       { vmr with API.vM_platform = platform }
     end else vmr
   in
 
+  (* Check to see if secure boot should be parsed *)
+  let pool = Helpers.get_pool ~__context in
+  if List.assoc "secureboot" platform = "auto" then
```

Comment:
The platform field is already updated above (see ensure_device_model_profile_present), so setting secureboot status should also be done there, perhaps by introducing a new function `|> update_platform_secureboot`, then you wouldn't need to modify the code that call Db.VM.set_platform.